### PR TITLE
handle null id [fix #71]

### LINKED ISSUE #71: Skip history if test case id is null

### PR DESCRIPTION
**Pull request by ehborisov**

### CODE REVIEW

File: allure-generator/src/main/java/io/qameta/allure/allure2/Allure2ResultsReader.java [hunk 1]
```diff
@@ -75,7 +77,12 @@ public Allure2ResultsReader(AttachmentsStorage storage) {
                     TestCaseResult dest = new TestCaseResult();
                     dest.setUid(generateUid());
 
-                    dest.setId(result.getId());
+                    String id = Optional.ofNullable(result.getId()).orElseGet(() -> {
```
Comment (baev) [inline allure-generator/src/main/java/io/qameta/allure/allure2/Allure2ResultsReader.java]: What is the point of generation random id here?

File: allure-generator/src/main/java/io/qameta/allure/history/HistoryResultAggregator.java [hunk 2]
```diff
@@ -35,22 +35,24 @@
...
-                    .withStatus(result.getStatus())
-                    .withStatusDetails(Objects.isNull(result.getFailure()) ? null : result.getFailure().getMessage())
-                    .withTime(result.getTime())
-                    .withTestRunName(testRun.getName());
-
-            List<HistoryItem> newItems = Stream.concat(Stream.of(newItem), data.getItems().stream())
-                    .limit(5)
-                    .collect(Collectors.toList());
-            data.setItems(newItems);
+            if(Objects.nonNull(result.getId())){
```
Comment (baev) [inline allure-generator/src/main/java/io/qameta/allure/history/HistoryResultAggregator.java]: to reduce complexity it is better to use return if id is null

File: allure-generator/src/main/java/io/qameta/allure/allure2/Allure2ResultsReader.java (hunk 1 above)
Comment (ehborisov) [inline allure-generator/src/main/java/io/qameta/allure/allure2/Allure2ResultsReader.java]: This is where test case result with null id is being parsed. We discussed that we wand to generate an id if it has not been provided, thus making this test an orphan, aren't we?

Comment (baev) [inline allure-generator/src/main/java/io/qameta/allure/allure2/Allure2ResultsReader.java]: > We discussed that we wand to generate an id if it has not been provided

Nope, that is gonna be a feature - use null id to disable history & statistics for test case

Comment (ehborisov) [inline allure-generator/src/main/java/io/qameta/allure/allure2/Allure2ResultsReader.java]: Ok, I made the corrections

**Comment by baev** [comment]
@ehborisov lets also update this line 
https://github.com/allure-framework/allure2/blob/master/allure-generator/src/main/java/io/qameta/allure/ProcessStage.java#L94

**Pull request merged**